Server/IT-Community.Server.Infrastructure/Services/CommentService.cs

```diff
@@ -0,0 +1,153 @@
+﻿using AutoMapper;
+using IT_Community.Server.Core.DataAccess;
+using IT_Community.Server.Core.Entities;
+using IT_Community.Server.Infrastructure.Dtos.CommentDTOs;
+using IT_Community.Server.Infrastructure.Dtos.PostDtos;
+using IT_Community.Server.Infrastructure.Exceptions;
+using IT_Community.Server.Infrastructure.Interfaces;
+using IT_Community.Server.Infrastructure.Resources;
+using Microsoft.AspNetCore.Identity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace IT_Community.Server.Infrastructure.Services
+{
+    public class CommentService : ICommentService
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        private readonly IMapper _mapper;
+        private readonly UserManager<User> _userManager;
+
+        public CommentService(IUnitOfWork unitOfWork, IMapper mapper, UserManager<User> userManager)
+        {
+            _unitOfWork = unitOfWork;
+            _mapper = mapper;
+            _userManager = userManager;
+        }
+
+        public List<CommentPostDto> GetCommentsWithReplies(int postId)
+        {
+            var commentList = _mapper.Map<List<CommentPostDto>>
+                (_unitOfWork.CommentRepository.GetAll(x => x.PostId == postId && x.ParentId == null, includeProperties: new[] {"Comments"}));
+            foreach(var comment in commentList)
+            {
+                LoadComments(comment, postId);
```

Comment:
But the replies should be already loaded by using Include()... or not? I think, you can avoid this loading cycle.

Reply:
They`re not loading automaticatilly, only first layer. BUt there can be much more, so we need to load them in this method